### PR TITLE
fix(webpack dev server): webpack dev server supports multiple webpack…

### PR DESCRIPTION
**Pull request by alan-agius4**
… compilers are configuration

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [] The package does not provide its own types, and you can not add them.
- [] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://webpack.js.org/configuration/node/>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/webpack-dev-server/index.d.ts [hunk 1]
```diff
@@ -70,20 +70,15 @@ declare namespace WebpackDevServer {
 
 declare class WebpackDevServer {
     constructor(
-        webpack: webpack.compiler.Compiler,
+        webpack: webpack.Compiler | webpack.MultiCompiler,
```
Comment (ZheyangSong) [inline types/webpack-dev-server/index.d.ts]: Probably `webpack.ICompiler`?

Comment (ZheyangSong) [reply types/webpack-dev-server/index.d.ts]: The suggestion is based on `webpack-dev-middleware`'s implementation and typing.

https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L82
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/webpack-dev-middleware/index.d.ts#L12

Comment (alan-agius4) [reply types/webpack-dev-server/index.d.ts]: No.

In webpack it is declared as such;
```
declare function webpack(options?: webpack.Configuration): webpack.Compiler;
```

Comment (alan-agius4) [reply types/webpack-dev-server/index.d.ts]: In webpack it is declared as such;
```
declare function webpack(options?: webpack.Configuration): webpack.Compiler;
```

`Compiler` extends `ICompiler` and just to mention `ICompiler` should be renamed as it's not following Typescript guidelines. As interfaces shouldn't not be prefixed with `I`

Comment (ZheyangSong) [reply types/webpack-dev-server/index.d.ts]: ICompiler is an interface. Compiler is a concrete class that implements the interface and extends Tapable. MultiComplier is an abstract class that implements the interface. If you check the implementation, the webpack dev server uses a method named plugin which is from Tapable which is inherited by Compiler only. MultiComplier doesn't guarantee to provide this method. I feel Compiler should be the best match. Not even the ICompiler I suggested... Again, this is a suggestion.

As of the guideline, it's for The Typescript project itself. Personally, I think allowing prefix 'I' to interface name is Ok. There are discussions about this guideline out there, such as https://stackoverflow.com/questions/31876947/confused-about-the-interface-and-class-coding-guidelines-for-typescript. I've been using TS for several projects and found that simple prefix like 'I' is helpful in identifying the significant code quickly. 🙂

Comment (alan-agius4) [reply types/webpack-dev-server/index.d.ts]: So `PR` approved :P?

Comment (ZheyangSong) [reply types/webpack-dev-server/index.d.ts]: yes

**Comment by alan-agius4** [comment]
@andy-ms,  seems like this hasn't been released

**Comment by alan-agius4** [comment]
@andy-ms seems like the merged changes havn’t been published. thanks

**Pull request merged**